### PR TITLE
Refactor fields (step 1)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Infrastructure "Refactor fields (step 1)"```

#### Purpose of change

Unhardcode data structures to allow field type customization/addition via json.

#### Describe the solution

Most of field related code was not touched in years, so changes would be massive. I will try to split it to somewhat sensible and reviewable chunks.

Current pull request does the following:

- reorganizes some field related code to make it easier to work with it later;
- adds code to load field type definitions from json;
- adds json definitions for existing field types (format is not final and is to be updated later).

#### Additional context

Players can notice that field types data is actually being loaded from json during init, but there are should be no game affecting changes as loaded data is used anywhere yet (just an additional small memory footprint).

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Hallelujah!

**Comment by nexusmrsep** [comment]
If you don't mind a design suggestion, could you perhaps by any chance unlimit the number of stages of the field so it would be  defined individually in JSON, so hot/cold air wouldn't need to be separated no more into separate incremental variants, and be streamlined as a singular field of many stages?

**Comment by ifreund** [comment]
Based on discussions earlier today I believe @ZhilkinSerg was already intending to that in fact, and that's part of what "format is not final and is to be updated later" means.

**Comment by ZhilkinSerg** [comment]
There are no limits on numbers of field intensity levels in json structure - you can store as many as you'd like. However, various code behind doing field processing only checks up to 3 intensity levels for existing field types - this would be changed later as more things would be refactored.

**Comment by BrettDong** [comment]
Which data entries in JSON need to be localized?

**Comment by ZhilkinSerg** [comment]
Probably only name field inside of intensity levels node, but it is not a final state of json object yet. Maybe I will change it later.

File: data/json/field_type.json [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+    "phase": "liquid",
+    "accelerated_decay": true,
+    "is_draw_field": true
+  },
+  {
+    "id": "fd_acid",
+    "type": "field_type",
+    "legacy_enum_id": 7,
+    "intensity_levels": [
+      { "name": "acit splatter", "sym": "5", "color": "light_green", "dangerous": true },
```
Comment (BevapDin) [inline data/json/field_type.json]: I'm not a native English speaker, is "acit" a real word or just a misspelling?

Comment (Zireael07) [reply data/json/field_type.json]: Probably was meant to be "acid"

Comment (ZhilkinSerg) [reply data/json/field_type.json]: Thanks! That was a typo.

image

**Pull request merged**